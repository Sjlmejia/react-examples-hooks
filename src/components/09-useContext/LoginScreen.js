import React, { useContext } from 'react'
import { UserContext } from './UserContext'

export const LoginScreen = () => {
  const {setUser} = useContext(UserContext)
  const updateUser = () => {
    setUser({
      id: 1234,
      name: 'Jorge'
    });
  };
  return (
    <div>
      <h1>Login Screen</h1>
      <hr />
      <button className="btn btn-primary" onClick={updateUser}>
        Login
      </button>
    </div>
  )
}
