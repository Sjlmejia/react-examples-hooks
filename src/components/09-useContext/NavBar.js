import React from 'react'
import { Link, NavLink } from 'react-router-dom'

export const NavBar = () => {
  return (
    <nav className="navbar navbar-expand-lg navbar-light bg-light">
      <Link to="/" className="navbar-brand">UseContext</Link>
      <ul className="navbar-nav me-auto mb-2 mb-lg-0">
        <li className="nav-item">
          <NavLink exact activeClassName="active" className="nav-link active" to="/">Home</NavLink>
        </li>
        <li className="nav-item">
          <NavLink exact activeClassName="active"  className="nav-link" to="/about">About</NavLink>
        </li>
        <li className="nav-item">
          <NavLink exact activeClassName="active" className="nav-link" to="/login">Login</NavLink>
        </li>
      </ul>
      
    </nav>
  )
}
