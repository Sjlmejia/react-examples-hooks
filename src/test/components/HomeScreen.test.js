import { mount, shallow } from 'enzyme';
import React from 'react'
import { HomeScreen } from '../../components/09-useContext/HomeScreen';
import { UserContext } from '../../components/09-useContext/UserContext';
describe('Pruebas en HomScreen', () => {
  const user = {
    name: 'Jorge',
    email:'jorge@gmail.com'
  }
  const wrapper = mount(
    <UserContext.Provider value ={{
      user
    }}>
      <HomeScreen />
    </UserContext.Provider>
  );
  test('debe crearse correctamente', () => {
    expect(wrapper).toMatchSnapshot();
  });
});