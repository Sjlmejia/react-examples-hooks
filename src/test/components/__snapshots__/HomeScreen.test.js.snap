// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en HomScreen debe crearse correctamente 1`] = `
<div>
  <h1>
    Home Screen
  </h1>
  <hr />
  <pre>
    {
    "name": "Jorge",
    "email": "jorge@gmail.com"
}
  </pre>
</div>
`;
